docs/proposals/runtime-client-server.md

```diff
@@ -0,0 +1,183 @@
+
+```proto
+// Runtime service defines the public APIs for remote container runtimes
+service RuntimeService {
+    // Version returns the runtime name, runtime version and runtime API version
+    rpc Version(VersionRequest) returns (VersionResponse) {}
+    // Status returns error if the runtime is unhealthy; nil otherwise
+    rpc Status(StatusRequest) returns (StatusResponse) {}
+
+    // CreatePodSandbox creates a pod-level sandbox. PodSandbox is interpreted 
+    // differently in different runtimes.  For docker, this may mean 
+    // creating the infra container. For hyper, this can create the 
+    // virtual machine.
+    rpc CreatePodSandbox(CreatePodSandboxRequest) returns (CreatePodSandboxResponse) {}
+    // DeletePodSandbox deletes the sandbox.
+    rpc DeletePodSandbox(DeletePodSandboxRequest) returns (DeletePodSandboxResponse) {}
+    // PodSandboxStatus returns the Status of the PodSandbox.
+    rpc PodSandboxStatus(PodSandboxStatusRequest) returns (PodSandboxStatusResponse) {}
+    // ListPodSandbox returns a list of SandBox.
+    rpc ListPodSandbox(ListPodSandboxRequest) returns (ListPodSandboxResponse) {}
+    // PortForward copies the data from the port in the PodSandBox to the stream.
+    rpc PortForward(stream PortForwardRequest) returns (stream PortForwardResponse) {}
+
+    // CreateContainer creates a new container in specified PodSandbox
+    rpc CreateContainer(CreateContainerRequest) returns (CreateContainerResponse) {}
+    // StartContainer starts the container.
+    rpc StartContainer(StartContainerRequest) returns (StartContainerResponse) {}
+    // StopContainer stops a running container with a grace period (i.e., timeout).
+    rpc StopContainer(StopContainerRequest) returns (StopContainerResponse) {}
+    // RemoveContainer removes the container.
+    rpc RemoveContainer(RemoveContainerRequest) returns (RemoveContainerResponse) {}
+    // ListContainers lists all containers by filters.
+    rpc ListContainers(ListContainersRequest) returns (ListContainersResponse) {}
+    // ContainerStatus returns the container status.
+    rpc ContainerStatus(ContainerStatusRequest) returns (ContainerStatusResponse) {}
+    // Exec execute a command in the container.
+    rpc Exec(stream ExecRequest) returns (stream ExecResponse) {}
+}
+
+// Image service defines the public APIs for managing images
+service ImageService {
+    // ListImages lists existing images.
+    rpc ListImages(ListImagesRequest) returns (ListImagesResponse) {}
+    // ImageStatus returns the status of the image.
+    rpc ImageStatus(ImageStatusRequest) returns (ImageStatusResponse) {}
+    // PullImage pulls a image with authentication config.
+    rpc PullImage(PullImageRequest) returns (PullImageResponse) {}
```

Comment:
I will want to charge the image pull to the pod level cgroup.

I need the ability to associate an image pull to a pod sandbox.

/cc @vishh per our chat.